sklearn/utils/extmath.py

```diff
@@ -696,20 +696,24 @@ def _incremental_mean_and_var(X, last_mean, last_variance, last_sample_count):
     # old = stats until now
     # new = the current increment
     # updated = the aggregated stats
-    last_sum = last_mean * last_sample_count
     new_sum = np.nansum(X, axis=0)
 
     new_sample_count = np.sum(~np.isnan(X), axis=0)
     updated_sample_count = last_sample_count + new_sample_count
 
-    updated_mean = (last_sum + new_sum) / updated_sample_count
+    new_mean = new_sum / new_sample_count
+    delta = new_mean - last_mean
+    updated_mean = \
+        last_mean + (delta * new_sample_count) / updated_sample_count  # fixes test
+        # (last_mean * last_sample_count + new_sum) / updated_sample_count  # breaks stability test
```

Comment:
this is line is how it was before and is necessary to have the stability test below to pass. However this line is the problem for the issue reported as `last_mean * last_sample_count` will explode.

So I feel a bit stuck. Either I change the test  or ... Any  thought?

Reply:
So you're saying `last_mean + (delta * new_sample_count) / updated_sample_count` is unstable, but avoids overflow?

Reply:
Some notes on this issue:

The mean in this PR causes the tolerance in the variance to go to ~458, which is up from the original 177. All it takes to get ~458 is by factoring out `last_sample_count`:

```
(last_mean + new_sum/last_sample_count)*last_sample_count / updated_sample_count
```

This PRs updated_means differ from master on the order of 10-9, which is enough to cause the instability in the variance. (The variance is on the order of 10+15)